core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -1556,7 +1555,7 @@ private boolean isIntervalHourAndSecond(SqlCall call) {
         return createSqlDataTypeSpecByName(typeAlias, typeName);
       case CHAR:
       case VARCHAR:
-        if (isColumnLengthPresent) {
+        if (isContainsPrecision) {
```

Comment:
this regular expression doesnt look right to me
boolean isContainsPrecision = type.toString().matches("^.*[\\(\\)].*");
can you quickly check?

Reply:
Done